modules/network-load-balancer/main.tf

```diff
@@ -0,0 +1,87 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# DEPLOY A NETWORK LOAD BALANCER
+# This module deploys a GCP Regional Network Load Balancer
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+terraform {
+  # This module has been updated with 0.12 syntax, which means it is no longer compatible with any versions below 0.12.
+  required_version = ">= 0.12"
+}
+
+# ------------------------------------------------------------------------------
+# CREATE FORWARDING RULE
+# ------------------------------------------------------------------------------
+
+resource "google_compute_forwarding_rule" "default" {
+  provider              = google-beta
+  project               = var.project
+  name                  = var.name
+  target                = google_compute_target_pool.default.self_link
+  load_balancing_scheme = "EXTERNAL"
+  port_range            = var.port_range
+  ip_address            = var.ip_address
+  ip_protocol           = var.protocol
+
+  labels = var.custom_labels
+}
+
+# ------------------------------------------------------------------------------
+# CREATE TARGET POOL
+# ------------------------------------------------------------------------------
+
+variable "instances" {
+  description = "List of self links to instances in the pool. Note that the instances need not exist at the time of target pool creation."
+  type        = list(string)
+  default     = []
+}
+
+resource "google_compute_target_pool" "default" {
+  project          = var.project
```

Comment:
Should we use `google-beta` on all the resources?

Reply:
I've used `google-beta` selectively in the resources where required. What advantage would we get by using `google-beta` across the line?

Reply:
In case users define specific projects / credentials (or other config) in the `google-beta` provider but not the `google` provider.

Reply:
Got it.